### PR TITLE
Extract DSA TitleBuilder into Cocina::Services::TitleBuilder

### PR DESCRIPTION
**Pull request by aaron-collier**
**NOTE:  Changes to openapi.yml require updating openapi.yml for sdr-api and dor-services-app and generating models - see README.**

## Why was this change made? 🤔

Part of #311 

## How was this change tested? 🤨

Added Unit Tests

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: lib/cocina/models/admin_policy.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ class AdminPolicy < Struct
 
       include Checkable
 
+      include TitleBuilder
```
Comment (jcoyne) [inline lib/cocina/models/admin_policy.rb:10]: I don't think we should include these behaviors on the models.  Can these just be services?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: lib/cocina/models/admin_policy.rb (hunk 1 above)
Comment (aaron-collier) [reply lib/cocina/models/admin_policy.rb:10]: Yeah - totally, I was actually thinking that would be a good idea but would like a short pairing to discuss.

**Comment by aaron-collier** [comment]
@jcoyne marking this as draft while I work on improving the test coverage and address a bug or two that has come up in doing so.

File: lib/cocina/services/title_builder.rb [hunk 2]
```diff
@@ -0,0 +1,203 @@
+# frozen_string_literal: true
+
+module Cocina
+  module Services
```
Comment (jcoyne) [inline lib/cocina/services/title_builder.rb:4]: I think this should still be `models` since this is the `cocina-models` gem.

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Review by jcoyne — APPROVED** [review]

**Pull request merged**